extensions/wikia/HubRssFeed/models/BaseRssModel.class.php

```diff
@@ -532,4 +545,43 @@ protected function findAvailableTimestamp( $timestamp, $uniqueTimestampsCount )
 		return $newTimestamp;
 	}
 
+	/**
+	 * Checks if response code is different than 404. WE don't care about 503 etc as they might change
+	 * @param $url
+	 * @return bool
+	 */
+	protected function checkURLExists( $url ) {
+		$req = Http::request( "HEAD", $url, [ 'returnInstance' => true, 'followRedirects' => true ] );
+		return $req->getStatus() != "404";
+	}
+
+	/**
+	 * Removes "dead" urls from DB
+	 * @param $feedName
+	 */
+	protected function cleanDeadUrlsInDB( $feedName ) {
+		$urlsMap = $this->getFeedData();
+		foreach ( $urlsMap as $url => $item ) {
+			if ( $item[ 'wikia_id' ] && $item[ 'page_id' ] && !$this->checkURLExists( $url ) ) {
+				$this->deleteRow( $item[ 'wikia_id' ], $item[ 'page_id' ], $feedName );
+			}
+		}
+	}
+
+	/**
+	 * Removes "dead" urls from array (keys)
+	 * @param $items
+	 * @return array
+	 */
+	protected function cleanDeadUrlsFromArray( $items ) {
```

Comment:
What is the difference between this and cleanDeadUrlsInDB (what is the rationale why we need both)